wolfcrypt/src/port/iotsafe/iotsafe.c

```diff
@@ -488,11 +493,28 @@ static int iotsafe_readfile(uint8_t *file_id, uint16_t file_id_sz,
                 return -1;
             }
             off += ret/2;
+#ifdef IOTSAFE_NO_GETDATA
+            if (XSTRNCMP(&resp[ret-4], "0000", 4) == 0) {
```

Comment:
Should this instead be always stripping any trailing `00` characters, not just 4 explicitly?

Reply:
This will strip all trailing zeros, this check is confirming whether the file ends in multiple 0 bytes so it knows the file has ended and needs to be stripped.
I wanted to guard against the case that the file has a single 00 byte that happens to be at the end of the chunk.